### ISSUE TITLE
QMPlay2 deadlocks when opening a YT search result in the browser during playback

### ISSUE THREAD

**Issue by RJVB** [zaps166/QMPlay2#825]
This is a weird one. I can open a YT search result in the browser just fine but when I do this while the video is playing back (presumably any video) QMPlay2 deadlocks. Only the audio continues playing, and waiting for the video to finish doesn't solve the issue.

It looks to be related to a defunct/zombie child process that remains once the deadlock has occurred.

I'm unable to do any post-mortem debugging: gdb will complain that the process is already being traced (by the parent process, so the calling shell or init). When I start QMPlay2 under a debugger the deadlock doesn't occur.

**Comment by zaps166** [comment]
Is it a regression in 24.12.23?

**Comment by RJVB** [comment]
No but I'll tell you in a bit if the latest version resolves things; I hadn't yet updated.

**Comment by zaps166** [comment]
Ok, but I don't have such problem - no deadlock on Arch Linux.

**Comment by RJVB** [comment]
Any idea what's with the "process already being traced" that prevents me from attaching a debugger?

**Comment by zaps166** [comment]
> Any idea what's with the "process already being traced" that prevents me from attaching a debugger?

`sudo sysctl kernel.yama.ptrace_scope=0` ?

**Comment by RJVB** [comment]
I don't have that setting and AFAIK I can attach to  all other running processes (that I own).

**Comment by zaps166** [comment]
I don't know, I don't have debugger problems on QMPlay2.

**Comment by RJVB** [comment]
With the latest version, when I open-in-browser while the video is playing:

```
[...]
[ass] Using font provider fontconfig
[opus @ 0x3f116c0] Could not update timestamps for skipped samples.

[1]+  Stopped                 QMPlay2
bertin@Bola:~/work/src/Scratch/KDE/KF5$ Running Firefox Developer Edition (/opt/firefox/firefox-browser https://www.youtube.com/watch?v=Ey2ouSNXkSA)

bertin@Bola:~/work/src/Scratch/KDE/KF5$ jobs -l
[1]+  4802 Child exited            QMPlay2
```

**Comment by zaps166** [comment]
Something is wrong with your system. It's just `QDesktopServices::openUrl()`.

**Comment by RJVB** [comment]
I don't know. It's true that I  normally build QMPlay2/Linux against a very lightly patched Qt 5.12 (but not at the QDesktopServices level!). I just made a build against stock Qt 5.15.2 and it indeed doesn't show the deadlock issue. But it also doesn't open anything in the browser, something I had already noticed before.

(No, it seems I can no longer update to whatever Qt 5.15.x is the newest...)

**Comment by RJVB** [comment]
> `sudo sysctl kernel.yama.ptrace_scope=0` ?

I have `# CONFIG_SECURITY_YAMA is not set` in all my kernel configs, surely reflecting a Ubuntu default. Which explains why I can attach to any other application that doesn't disable debugging itself (like keepassxc does by default).

**Comment by RJVB** [comment]
OK, last question. Is there an easy way that I can check if a video is playing from within an extension module source? I think the deadlocking must be something Qt 5.12[.6] specific as I don't get it in a build against Qt 5.9 .

**Comment by RJVB** [comment]
OOOOKKKKKK

I finally had the idea to check if an application can asked to be traced, and indeed it can. So I set a breakpoint on `ptrace()` and discovered it was indeed being called from a function in `$prefix/lib/swiftshader/libvk_swiftshader.so`. Apparently I experimented with building swiftshader for some reason, and left that directory in my parallel prefix (and indeed a debug swiftshader build calls `ptrace(PTRACE_TRACEME,...)`).

This explains both the debugger not attaching, the deadlocking and presumably also a long-standing never-understood issue where QMPlay2 would suddenly behave as if it had received a SIGSTOP, including giving back the prompt when run in the foreground in a shell. At least the 1st two were solved my moving that swiftshader directory out of the way.

Sorry for the noise!
